RadixWallet/Features/AssetsFeature/Components/HideAsset/HideAsset.swift

```diff
@@ -0,0 +1,120 @@
+// MARK: - HideAsset
+@Reducer
+public struct HideAsset: Sendable, FeatureReducer {
+	@ObservableState
+	public struct State: Sendable, Hashable {
+		let asset: AssetAddress
+		var shouldShow = true
+
+		@Presents
+		var destination: Destination.State? = nil
+
+		public init(asset: AssetAddress) {
+			self.asset = asset
+		}
+	}
+
+	public typealias Action = FeatureAction<Self>
+
+	public enum InternalAction: Sendable, Equatable {
+		case setShouldShow(Bool)
+	}
+
+	public enum ViewAction: Sendable, Equatable {
+		case task
+		case buttonTapped
+	}
+
+	public enum DelegateAction: Sendable, Equatable {
+		case didHideAsset
+	}
+
+	// MARK: - Destination
+	public struct Destination: DestinationReducer {
```

Comment:
Did you try using the new style, where you just define something like:
```
extension ContactsFeature {
  @Reducer(state: .hashable, action: .equatable)
  enum Destination {
    case confirmation(ConfirmationAction)
  }
}
```

Reply:
how do you make the parent conform to `FeatureReducer` ?

Reply:
You can do put `@Reducer(state: .hashable, action: .equatable)` on the Destination if that is what you have in mind? But it was a genuine question, it seems that it only will play nicely with FeatureReducer sometimes.